ClientRuntimes/CSharp/Microsoft.Rest.ClientRuntime/BatchDelegatingHandler.cs

```diff
@@ -0,0 +1,185 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Rest
+{
+    /// <summary>
+    /// HTTP batch handler.
+    /// It expects the caller to dispose HttpRequestMessage and HttpResponseMessage.
+    /// It assumes the web service is supporting batch requests as per this MSDN article:
+    /// https://blogs.msdn.microsoft.com/webdev/2013/11/01/introducing-batch-support-in-web-api-and-web-api-odata/
+    /// i.e. requests are batched into a single request inside the multipart content.
+    /// </summary>
+    public class BatchDelegatingHandler : DelegatingHandler
+    {
+        /// <summary>
+        /// HTTP method (e.g. POST or PUT) to use in issuing a batch call.
+        /// </summary>
+        private readonly HttpMethod batchMethod;
+
+        /// <summary>
+        /// URL to issue the batch call to.
+        /// </summary>
+        private readonly Uri batchURL;
+
+        /// <summary>
+        /// List of operations to issue in a batch.
+        /// </summary>
+        private List<HttpRequestMessage> requests;
+
+        /// <summary>
+        /// Array of HTTP content returned by the server in response to operations that are issued in a batch.
+        /// </summary>
+        private HttpContent[] responses;
+
+        /// <summary>
+        /// Event that tells all tasks that they can proceed.
+        /// </summary>
+        private ManualResetEvent batchCompleted;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BatchDelegatingHandler"/> class.
+        /// </summary>
+        /// <param name="method">the type of method that the Batch operation is</param>
+        /// <param name="url">the URL that the Batch operation needs to hit</param>
+        public BatchDelegatingHandler(HttpMethod method, Uri url)
```

Comment:
I'll remove the ability to set the HTTP method and make it a POST. However, with the URI, my read of the OData spec indicates that it can include the service root, and getting that out of a requestUri is not trivial. How should we address that?